File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/product/ProductCreateDto.java
package com.intexsoft.webshop.productservice.dto.product;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.intexsoft.webshop.productservice.dto.attributevalue.AttributeValueCreateDto;
import com.intexsoft.webshop.productservice.dto.image.ImageCreateDto;
import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class ProductCreateDto {
    @NotBlank
    @Size(max = 255)
    String name;
    @NotNull
    @Positive
    Long subcategoryId;
    @NotNull
    @Positive
    Long vendorId;
    @Nullable
    @Size(min = 1)
    @JsonProperty("images")
    List<ImageCreateDto> imageCreateDtos;
    @Nullable
    @Size(min = 1)
    @JsonProperty("attributeValues")
    List<AttributeValueCreateDto> attributeValueCreateDtos;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/product/ProductCreateDto.kt
package com.intexsoft.webshop.productservicekt.dto.product

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.intexsoft.webshop.productservicekt.dto.attributevalue.AttributeValueCreateDto
import com.intexsoft.webshop.productservicekt.dto.image.ImageCreateDto
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.Size

@JsonIgnoreProperties(ignoreUnknown = true)
data class ProductCreateDto(
    @field:NotBlank
    @field:Size(max = 255)
    val name: String?,
    @field:Positive
    val subcategoryId: Long,
    @field:Positive
    val vendorId: Long,
    @JsonProperty("images")
    @field:Valid
    @field:Size(min = 1)
    val imageCreateDtos: List<ImageCreateDto>?,
    @JsonProperty("attributeValues")
    @field:Valid
    @field:Size(min = 1)
    val attributeValueCreateDtos: List<AttributeValueCreateDto>?
)